ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs

```diff
@@ -807,6 +807,8 @@ data StdRunNodeArgs m blk (p2p :: Diffusion.P2P) = StdRunNodeArgs
   , srnMaybeMempoolCapacityOverride :: Maybe MempoolCapacityBytesOverride
     -- ^ Determine whether to use the system default mempool capacity or explicitly set
     -- capacity of the mempool.
+  , srnChainSyncTimeout             :: Maybe NTN.ChainSyncTimeout
+    -- ^ A custom timeout for ChainSync.
```

Comment:
Given that `stdChainSyncTimeout` uses `IO` for randomisation
```haskell
stdChainSyncTimeout :: IO ChainSyncTimeout
```
do we want to take a `Maybe (m NTN.ChainSyncTimeout)` here? Or is it clear that this extra generality is unnecessary?

Reply:
See #509